artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQSession.java

```diff
@@ -397,6 +380,55 @@ public MessageProducer createProducer(final Destination destination) throws JMSE
       }
    }
 
+   void checkDestination(ActiveMQDestination destination) throws JMSException {
```

Comment:
make it final to avoid overriding it (it shouldn't happen eh)

Reply:
TBH I hate final methods...

I have needed quite a few times to override them on test cases... so I only reserve the word final where there is a strong "business" case to not let it be overwritten.  but that's jut my opinion.

Reply:
+1 same here not the biggest fan of them either, i remember having on another project starting with the name K ;), such issue, and getting people to agree to make it non final was a nightmare.

Reply:
I understand it and I agree, but depends on the visibility/importance of the method: this one if overridden really breaks badly pretty much everything.
I won't never use final for private methods, but for package private ie near-public and important ones, yes. 
As I've said is just style comment, I'm already happy with the PR :+1: